### PR TITLE
Update to Ofono 1.26

### PR DESCRIPTION
**Pull request by piggz**
This branch cherry-picks all the relevant commits from upstream ofono between 1.25 and 1.26, missing out only 2 or 3 related to ELL and functionality that is already in the sailfish ofono.

### CODE REVIEW

**Comment by monich** [comment]
I noticed that (at least) these commits were left out:

https://git.kernel.org/pub/scm/network/ofono/ofono.git/commit/?id=a5bdf48c
https://git.kernel.org/pub/scm/network/ofono/ofono.git/commit/?id=bd6f8078

Was that done on purpose?

**Comment by piggz** [comment]
Yes, mainly because of this in jolla ofono https://github.com/sailfishos/ofono/commit/f580867c12c3f2e8248c60084b449d48e2f1a1d7

File: ofono/include/types.h [hunk 1]
```diff
@@ -183,6 +183,28 @@ struct ofono_uuid {
...
+	OFONO_GPRS_PROTO_IP = 0,
+	OFONO_GPRS_PROTO_IPV6,
+	OFONO_GPRS_PROTO_IPV4V6,
+};
+
+enum ofono_gprs_auth_method {
+    OFONO_GPRS_AUTH_METHOD_ANY = 0,
+	OFONO_GPRS_AUTH_METHOD_CHAP,
+	OFONO_GPRS_AUTH_METHOD_PAP,
+	OFONO_GPRS_AUTH_METHOD_NONE,
```
Comment (monich) [inline ofono/include/types.h:205]: This is an interface and therefore enum values must not change (even if they no longer match the upstream)
```c
enum ofono_gprs_auth_method {
	OFONO_GPRS_AUTH_METHOD_ANY = 0,
	OFONO_GPRS_AUTH_METHOD_NONE,
	OFONO_GPRS_AUTH_METHOD_CHAP,
	OFONO_GPRS_AUTH_METHOD_PAP
};
```

File: ofono/src/common.c [hunk 2]
```diff
@@ -787,3 +791,64 @@ const char *ofono_access_technology_to_string(enum ofono_access_technology tech)
...
+		*proto = OFONO_GPRS_PROTO_IPV4V6;
+		return TRUE;
+	}
+
+	return FALSE;
+}
+
+const char *gprs_auth_method_to_string(enum ofono_gprs_auth_method auth)
+{
+	switch (auth) {
```
Comment (monich) [inline ofono/src/common.c:827]:
```
src/common.c: In function ‘gprs_auth_method_to_string’:
src/common.c:827:2: warning: enumeration value ‘OFONO_GPRS_AUTH_METHOD_ANY’ not handled in switch [-Wswitch]
  827 |  switch (auth) {
      |  ^~~~~~
```
It needs to be handled and `gprs_auth_method_from_string()` below has to be adjusted accordingly, too.

File: ofono/drivers/atmodem/atutil.c [hunk 3]
```diff
@@ -654,3 +655,45 @@ int at_util_get_ipv4_address_and_netmask(const char *addrnetmask,
 
 	return ret;
 }
+
+int at_util_gprs_auth_method_to_auth_prot(
+					enum ofono_gprs_auth_method auth_method)
+{
+	switch (auth_method) {
```
Comment (monich) [inline ofono/drivers/atmodem/atutil.c:662]: This warning should be silenced too, somehow.
```
drivers/atmodem/atutil.c: In function ‘at_util_gprs_auth_method_to_auth_prot’:
drivers/atmodem/atutil.c:662:2: warning: enumeration value ‘OFONO_GPRS_AUTH_METHOD_ANY’ not handled in switch [-Wswitch]
  662 |  switch (auth_method) {
      |  ^~~~~~
```

**Comment by monich** [comment]
> Yes, mainly because of this in jolla ofono [f580867](https://github.com/sailfishos/ofono/commit/f580867c12c3f2e8248c60084b449d48e2f1a1d7)

From a quick glance those wouldn't harm but you may know better.

File: ofono/src/common.c (hunk 2 above)
Comment (monich) [reply ofono/src/common.c:827]: The same way as it was done in `src/gprs.c`

**Comment by monich** [comment]
I'd say it's fine to add new commits on top of these, it that saves you time.

**Comment by monich** [comment]
Or I can make those changes myself

**Comment by monich** [comment]
Smoke-tested, nothing seems to be broken from the first glance.

**Comment by monich** [comment]
Merged, thanks!

**Pull request merged**